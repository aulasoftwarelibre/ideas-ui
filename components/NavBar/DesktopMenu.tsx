import classnames from 'classnames';
import React from 'react';

import { menuLinks } from '../../config/links';
import DesktopLink from './DesktopLink';

export const DesktopMenu: React.FunctionComponent = ({}) => {
  return (
    <div className="hidden sm:block sm:ml-6">
      <div className="flex">
        {menuLinks.map((link) => (
          <DesktopLink
            key={link.path}
            label={link.label}
            path={link.path}
          ></DesktopLink>
        ))}
      </div>
    </div>
  );
};

export default DesktopMenu;
