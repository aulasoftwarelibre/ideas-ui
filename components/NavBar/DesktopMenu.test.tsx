import React from 'react';

import { renderMockRouter } from '../../utils/test-utils';
import DesktopMenu from './DesktopMenu';

describe("DesktopMenu", () => {
  it("should be rendered without active link", () => {
    const { asFragment } = renderMockRouter(<DesktopMenu />, {
      router: { pathname: "/no-active" },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be rendered with active link", () => {
    const { asFragment } = renderMockRouter(<DesktopMenu />, {
      router: { pathname: "/" },
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
