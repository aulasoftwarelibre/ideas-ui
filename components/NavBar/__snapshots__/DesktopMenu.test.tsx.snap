// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopMenu should be rendered with active link 1`] = `
<DocumentFragment>
  <div
    class="hidden sm:block sm:ml-6"
  >
    <div
      class="flex"
    >
      <span
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-ideaOrange"
      >
        Inicio
      </span>
      <a
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange"
        href="/groups"
      >
        Grupos
      </a>
      <a
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange"
        href="/help"
      >
        Ayuda
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DesktopMenu should be rendered without active link 1`] = `
<DocumentFragment>
  <div
    class="hidden sm:block sm:ml-6"
  >
    <div
      class="flex"
    >
      <a
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange"
        href="/"
      >
        Inicio
      </a>
      <a
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange"
        href="/groups"
      >
        Grupos
      </a>
      <a
        class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange"
        href="/help"
      >
        Ayuda
      </a>
    </div>
  </div>
</DocumentFragment>
`;
